### ISSUE TITLE
osrm_datastore does not work if osrm_map action is :create

### ISSUE THREAD

**Issue by tatchiuleung** [chr4-cookbooks/osrm#9]
Here is my recipe:
```
include_recipe 'apt'
include_recipe 'osrm'

region = 'delaware'

group 'osrm-routed'
user 'osrm-routed' do
  gid 'osrm-routed'
  shell '/bin/false'
end

osrm_map region do
  action :create
end

osrm_datastore region

osrm_routed region do
  listen '0.0.0.0'
  port   5000
  shared_memory true
end
```

And when I run kitchen converge:
```
================================================================================
           Error executing action `create` on resource 'osrm_datastore[delaware]'
           ================================================================================

           Mixlib::ShellOut::ShellCommandFailed
           ------------------------------------
           execute[/opt/osrm/build/osrm-datastore /opt/osrm-data/delaware/car/delaware-latest.osrm] (/tmp/kitchen/cache/cookbooks/osrm/providers/datastore.rb line 59) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
           ---- Begin output of /opt/osrm/build/osrm-datastore /opt/osrm-data/delaware/car/delaware-latest.osrm ----
           STDOUT:
           STDERR: [warn] Invalid file path given!
```

If I delete ```/opt/osrm-data/delaware/car``` and change the osrm_map action to :create_if_missing, kitchen converge works fine.

It looks like some files are deleted inadvertently from osrm_contract causing osrm_datastore to fail.

**Comment by chr4** [comment]
Thanks for your report!

It seems one of those locations need some adaption since the osrm-5.0.0 release. I think you're right and one of the following lines need adaption:

https://github.com/chr4-cookbooks/osrm/blob/master/providers/map_contract.rb#L36-L41
https://github.com/chr4-cookbooks/osrm/blob/master/providers/map_extract.rb#L59-L64

Can you try out whether the problem persists when you remove/ modify the sections above?

**Comment by chr4** [comment]
This should be fixed in `5.2.0`. Note that the `_if_missing` actions were deprecated, and the default actions do not delete (unused) files anymore.

Please reopen if the problem persists.
